Lib/test/test_functools.py

```diff
@@ -746,11 +746,8 @@ def wrapper():
         self.assertEqual(wrapper.attr, 'This is a different test')
         self.assertEqual(wrapper.dict_attr, f.dict_attr)
 
-@unittest.skipUnless(c_functools, 'requires the C _functools module')
-class TestReduce(unittest.TestCase):
-    if c_functools:
-        func = c_functools.reduce
 
+class TestReduce:
```

Comment:
Can you please rename it to BaseTestReduce to make it more obvious that it's not a regular test case?

Reply:
Done this way to be consistent with the other tests.

I have no objection to this suggestion, but as it would also need to be done for `TestPartial` and `TestCmpToKey`, I feel it is out of scope for this change.